jobs/job_options.py

```diff
@@ -0,0 +1,73 @@
+# coding: utf-8
+#
+# Copyright 2021 The Oppia Authors. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS-IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""Option class for configuring the behavior of Oppia jobs."""
+
+from __future__ import absolute_import  # pylint: disable=import-only-modules
+from __future__ import unicode_literals  # pylint: disable=import-only-modules
+
+import feconf
+
+import apache_beam as beam
+from apache_beam.options import pipeline_options
+
+
+class JobOptions(pipeline_options.GoogleCloudOptions):
+    """Option class for configuring the behavior of Oppia jobs."""
+
+    JOB_OPTIONS = {
+        'model_getter': (
+            beam.PTransform,
+            'The type of PTransform the pipeline should use to fetch models '
+            'from the datastore.'),
+    }
+
+    def __init__(self, flags=None, **job_options):
+        """Initializes a new JobOptions instance.
+
+        Args:
+            flags: dict(str:str)|None. Command-line flags for customizing a
+                pipeline. Although Oppia doesn't use command-line flags to
+                control jobs or pipelines, we still need to pass the value
+                (unmodified) because PipelineOptions, a parent class, needs it.
+            **job_options: dict(str: *). One of the options defined in the class
+                JOB_OPTIONS dict.
+        """
+        unsupported_options = set(job_options).difference(self.JOB_OPTIONS)
+        if unsupported_options:
+            unsupported_options = ', '.join(sorted(unsupported_options))
+            raise ValueError('Unsupported option(s): %s' % unsupported_options)
+        super(JobOptions, self).__init__(
+            # PipelineOptions: implementation detail.
```

Comment:
Bit confused about this. Are "PipelineOptions" and "GoogleCloudOptions" actual concepts/classes? If not, perhaps write these notes in natural language.

Reply:
No, they are parent classes that we're inheriting from. They have their own "options" that we need to fill to get the pipelines configured correctly.

Reply:
Where are we inheriting them from? I don't actually see them in the impl so I was a bit confused.

Reply:
Ah, it's a class. You can see on line 28 we're inheriting from `pipeline_options.GoogleCloudOptions`.

Reply:
Ah, I see, sorry missed that. Maybe say "Needed for the PipelineOptions superclass." and ditto below. (I don't know how to interpret "implementation detail".)

Reply:
Done.